test-utils/test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/random/IntegerGenerator.java

```diff
@@ -1,15 +1,64 @@
 package uk.gov.justice.services.test.utils.core.random;
 
+import static java.lang.String.format;
+
+import io.netty.util.internal.ThreadLocalRandom;
+
 public class IntegerGenerator extends Generator<Integer> {
 
-    private final Integer max;
+    /**
+     * The minimum value of the range
+     */
+    private final int min;
 
-    public IntegerGenerator(final Integer max) {
+    /**
+     * The maximum value of the range
+     */
+    private final int max;
+
+    /**
+     * Package Access only
+     * 
+     * @see RandomGenerator
+     */
+    IntegerGenerator() {
+        this.min = Integer.MIN_VALUE;
+        this.max = Integer.MAX_VALUE;
+    }
+
+    /**
+     * Package Access only
+     * 
+     * @param max value excluded
+     * @see RandomGenerator
+     */
+    IntegerGenerator(final int max) {
+        this.min = 0;
+        this.max = max;
+    }
+    /**
+     * Package Access only
+     * 
+     * @param min value included
+     * @param max value excluded
+     * @see RandomGenerator
+     */
+    IntegerGenerator(final int min, final int max) {
+        if (min >= max) {
+            throw new IllegalArgumentException(
+                            format("%s cannot be greater or equal to %s", min, max));
+        }
+        this.min = min;
         this.max = max;
     }
 
     @Override
     public Integer next() {
-        return RANDOM.nextInt(max);
+        if(min >= 0){
+            return ThreadLocalRandom.current().nextInt(min, max);
```

Comment:
Any benefit of using ThreadLocalRandom.current().nextInt(min, max) over RANDOM.ints(min, max).limit(1).findFirst().getAsInt(). The reason I am mentioning is we already have RANDOM as part of the parent class and it's easy to mock the RANDOM in testing, if required.